dask/dataframe/io/parquet/arrow.py

```diff
@@ -213,10 +214,16 @@ def _write_partitioned(
     fs.mkdirs(root_path, exist_ok=True)
 
     df = table.to_pandas(ignore_metadata=True)
+    index_cols = list(index_cols)
```

Comment:
Is there a chance that `index_cols` is None when it gets here? Or is it always reassigned in `write_partition`?

Reply:
Good point - It will never be None (it is reset to `[]` in `write_partition`, like you said), but it probably makes sense to be a bit more careful here.